### PR TITLE
Changelog feature

### PR DESCRIPTION
**Pull request by GeertvanHorrik**
### Description of Change ###

Feature allows any component to provide items to be visible in the changelog.

image

The changelog has a snapshot feature that shows the delta between the last time the software was used and now.

The use, a library only has to create a type that implements `IChangelogProvider` or `ChangelogProviderBase`:

```
public class ChangelogProvider : Orchestra.Changelog.ChangelogProviderBase
{
    public override async Task<IEnumerable<ChangelogItem>> GetChangelogAsync()
    {
        return new[]
        {
            new ChangelogItem
            {
                Group = "General",
                Name = "Changelog feature",
                Description = "The changelog feature will show a changelog of all components inside the application",
                Type = ChangelogType.Feature
            },
            new ChangelogItem
            {
                Group = "General",
                Name = "Bug example",
                Type = ChangelogType.Bug
            },
            new ChangelogItem
            {
                Group = "General",
                Name = "Change example",
                Type = ChangelogType.Change
            },
            new ChangelogItem
            {
                Group = "General",
                Name = "Improvement example",
                Type = ChangelogType.Improvement
            }
        };
    }
}
```

### Issues Resolved ### 

- fixes #

### API Changes ###

 
None

### Platforms Affected ### 

- All
- WPF
- UWP
- iOS
- Android

### Behavioral Changes ###

None

### PR Checklist ###

- [ ] I have included examples or tests
- [ ] I have updated the change log
- [ ] I am listed in the CONTRIBUTORS file
- [ ] Rebased on top of the target branch at time of PR
- [ ] Changes adhere to coding standard

### CODE REVIEW

File: src/Orchestra.Core/Changelog/Services/Interfaces/IChangelogService.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿namespace Orchestra.Changelog
+{
+    using System.Collections.Generic;
```
Comment (ceo1647) [inline src/Orchestra.Core/Changelog/Services/Interfaces/IChangelogService.cs:3]: unused directive

File: src/Orchestra.Core/Changelog/ViewModels/ChangelogViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            _changelogService = changelogService;
+            _changelogSnapshotService = changelogSnapshotService;
+
+            Items = changelog.Items;
+            Title = changelog.Title ?? LanguageHelper.GetString("Orchestra_Changelog");
+        }
+
+        public Changelog Changelog { get; }
+
+        public List<ChangelogItem> Items { get; private set; }
```
Comment (ceo1647) [inline src/Orchestra.Core/Changelog/ViewModels/ChangelogViewModel.cs:32]: Items never changed in class maybe just {get;}

File: src/Orchestra.Core/ViewModels/AboutViewModel.cs [hunk 3]
```diff
@@ -17,27 +17,32 @@ namespace Orchestra.ViewModels
     using Catel.Services;
     using Catel.Reflection;
```
Comment (ceo1647) [inline src/Orchestra.Core/ViewModels/AboutViewModel.cs:18]: unused

File: src/Orchestra.Examples.Ribbon.Fluent/Changelog/Providers/ChangelogProvider.cs [hunk 4]
```diff
@@ -0,0 +1,57 @@
+﻿namespace Orchestra.Examples.Ribbon.Changelog.Providers
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+    using Orchestra.Changelog;
+
+    public class ChangelogProvider : Orchestra.Changelog.ChangelogProviderBase
```
Comment (ceo1647) [inline src/Orchestra.Examples.Ribbon.Fluent/Changelog/Providers/ChangelogProvider.cs:9]: just ChangelogProviderBase

File: src/Orchestra.Examples.Ribbon.Fluent/Changelog/Providers/ChangelogProvider.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
+﻿namespace Orchestra.Examples.Ribbon.Changelog.Providers
+{
+    using System;
```
Comment (ceo1647) [inline src/Orchestra.Examples.Ribbon.Fluent/Changelog/Providers/ChangelogProvider.cs:3]: not used

File: src/Orchestra.Examples.Ribbon.Fluent/Changelog/Providers/ChangelogProvider.cs [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    using System.Collections.Generic;
+    using System.Text;
```
Comment (ceo1647) [inline src/Orchestra.Examples.Ribbon.Fluent/Changelog/Providers/ChangelogProvider.cs:5]: not used

File: src/Orchestra.Examples.Ribbon.Fluent/Changelog/Providers/ChangelogProvider.cs [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+                },
+                new ChangelogItem
+                {
+                    Group = "General",
+                    Name = "Improvement example",
+                    Type = ChangelogType.Improvement
+                }
+            });
+
+            for (int i = 0; i < 100; i++)
```
Comment (ceo1647) [inline src/Orchestra.Examples.Ribbon.Fluent/Changelog/Providers/ChangelogProvider.cs:44]: var i = 0

File: src/Orchestra.Tests/Changelog/Models/Extensions/ChangelogExtensionsFacts.cs [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+    public class ChangelogExtensionsFacts
+    {
+        [TestFixture]
+        public class TheGetDeltaMethod
+        {
+            [TestCase]
+            public void ReturnsEmptySnapshot()
+            {
+                var changelog1 = new Orchestra.Changelog.Changelog();
```
Comment (ceo1647) [inline src/Orchestra.Tests/Changelog/Models/Extensions/ChangelogExtensionsFacts.cs:14]: just Changelog and in a cases below

**Review by ceo1647 — COMMENTED** [review]
all looks good, just a little fixes mostly in using directives

File: src/Orchestra.Core/Changelog/Models/Changelog.cs [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+    public class Changelog
+    {
+        public Changelog()
+        {
+            Items = new List<ChangelogItem>();
+        }
+
+        public string Title { get; set; }
+
+        public List<ChangelogItem> Items { get; private set; }
```
[the remark this answers is not in the document]
Comment (GeertvanHorrik) [reply src/Orchestra.Core/Changelog/Models/Changelog.cs:15]: Code habit from before we could write private setters without a setter.

File: src/Orchestra.Core/Changelog/Markup/ChangelogTypeIcon.cs [hunk 10]
```diff
@@ -0,0 +1,59 @@
+﻿namespace Orchestra.Changelog.Markup
+{
+    using System;
+    using System.Windows;
+    using System.Windows.Data;
+
+    public class ChangelogTypeIcon : Catel.Windows.Markup.UpdatableMarkupExtension
```
[the remark this answers is not in the document]
Comment (GeertvanHorrik) [reply src/Orchestra.Core/Changelog/Markup/ChangelogTypeIcon.cs:7]: Checked code here:

https://github.com/dotnet/wpf/blob/89d172db0b7a192de720c6cfba5e28a1e7d46123/src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/DynamicResourceExtension.cs

**Pull request merged**